client/my-sites/email/controller.js

```diff
@@ -86,6 +87,17 @@ export default {
 		next();
 	},
 
+	emailManagementTitanSetupThankYouPage( pageContext, next ) {
+		pageContext.primary = (
+			<TitanSetupThankYou
+				domainName={ pageContext.params.domain }
+				emailAddress={ pageContext.query.email }
```

Comment:
The `TitanSetupThankYou` component is currently designed to retrieve the email address as a query parameter:

https://wordpress.com/email/:domain/titan/thank-you/:site?email=:email

Since `:email` should always point to the same domain than `:domain`, what if asked for the mailbox instead?

https://wordpress.com/email/:domain/titan/thank-you/:site?mailbox=:mailbox

We can indeed build the email address from these two pieces of information inside the `TitanSetupThankYou` component (i.e. `:mailbox@:domain`). We could even add that mailbox to the path so the url is even shorter:

https://wordpress.com/email/:domain/titan/thank-you/:mailbox/:site

That would give us something like:

https://wordpress.com/email/example.com/titan/thank-you/contact/example.wordpress.com

Reply:
Thinking about it more, since we will introduce the [new page](https://github.com/Automattic/wp-calypso/pull/54934) for setting up the unused mailbox at:

https://wordpress.com/email/:domain/titan/setup/:site

Or at - depending on the outcome of [this discussion](https://github.com/Automattic/wp-calypso/pull/54934#discussion_r678466357):

https://wordpress.com/email/:domain/titan/setup-mailbox/:site

It probably makes sense to move this new `Thank You` page at:

https://wordpress.com/email/:domain/titan/setup/thank-you/:mailbox/:site